extras/basic_example/basicServer.js

```diff
@@ -84,19 +84,19 @@ var deleteRoomsIfEmpty = function (theRooms, callback) {
         return;
     }
     var theRoom = theRooms.pop();
-    N.API.getUsers(theRoom._id, function(userlist){
+    N.API.getUsers(theRoom._id, function(userlist) {
         var users = JSON.parse(userlist);
         if (Object.keys(users).length === 0){
             N.API.deleteRoom(theRoom._id, function(){
-                if (theRooms.length > 0){
-                    deleteRoomsIfEmpty(theRooms, callback);
-                }
+                deleteRoomsIfEmpty(theRooms, callback);
             });
         } else {
-            if (theRooms.length > 0){
-                deleteRoomsIfEmpty(theRooms, callback);
-            }
+            deleteRoomsIfEmpty(theRooms, callback);
         }
+    }, function () {
```

Comment:
The error callback may very well be caused by a timeout, we either don't handle it or we should take into account what kind of errors it will report.
Consider the possible errors, if the room does not exist then deleting it will also fail, if it's a timeout (nuve is not available) then it also won't work. I don't see the value in the callback as it is.